pkg/server/server.go

```diff
@@ -189,12 +219,20 @@ func NewServerWithOpts(opts ...OpenFGAServiceV1Option) (*Server, error) {
 		maxConcurrentReadsForCheck:       defaultMaxConcurrentReadsForCheck,
 		maxConcurrentReadsForListObjects: defaultMaxConcurrentReadsForListObjects,
 		experimentals:                    make([]ExperimentalFeatureFlag, 0, 10),
+
+		checkQueryCacheEnabled: defaultCheckQueryCacheEnable,
+		checkQueryCacheLimit:   defaultCheckQueryCacheLimit,
+		checkQueryCacheTTL:     defaultCheckQueryCacheTTL,
 	}
 
 	for _, opt := range opts {
 		opt(s)
 	}
 
+	s.checkCache = ccache.New(
+		ccache.Configure[*graph.ResolveCheckResponse]().MaxSize(int64(s.checkQueryCacheLimit)),
+	)
```

Comment:
Someone that opts out of this new feature will not want the memory usage increase